src/extensions/errors.rs

```diff
@@ -0,0 +1,151 @@
+//! # Extension errors.
+//!
+//! An `ExtensionError` is thrown when an extension is invalid (for example when
+//! decoding from raw bytes) or when a check on an extension fails.
+//!
+//! `ExtensionError` holds individual errors for each extension.
+//! * `CapabilitiesExtensionError`
+//! * `LifetimeExtensionError`
+//! * `KeyPackageIdError`
+//! * `ParentHashError`
+//! * `RatchetTreeError`
+
+use crate::{codec::CodecError, config::ConfigError};
+use std::error::Error;
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[repr(u16)]
+pub enum ExtensionError {
+    /// Invalid extension type error.
+    InvalidExtensionType,
+
+    /// Error when decoding an extension.
+    DecodingError,
+
+    /// Capabilities extension error.
+    /// See `CapabilitiesExtensionError` for details.
+    Capabilities(CapabilitiesExtensionError),
+
+    /// Lifetime extension error.
+    /// See `LifetimeExtensionError` for details.
+    Lifetime(LifetimeExtensionError),
+
+    /// Key package ID extension error.
+    /// See `KeyPackageIdError` for details.
+    KeyPackageId(KeyPackageIdError),
+
+    /// Parent hash extension error.
+    /// See `ParentHashError` for details.
+    ParentHash(ParentHashError),
+
+    /// Ratchet tree extension error.
+    /// See `RatchetTreeError` for details.
+    RatchetTree(RatchetTreeError),
+}
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[repr(u16)]
+pub enum LifetimeExtensionError {
+    /// Invalid lifetime extensions.
+    Invalid = 0,
+
+    /// Lifetime extension is expired.
+    Expired = 1,
+}
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+#[repr(u16)]
+pub enum CapabilitiesExtensionError {
+    /// Invalid capabilities extensions.
+    Invalid = 0,
+
+    /// Capabilities extension is missing a version field.
+    EmptyVersionsField = 1,
```

Comment:
Could there be unsupported versions?

Reply:
Yep, but nothing we check right now. I updated #83 to follow up on these.